### ISSUE TITLE
failover cluster1 -> cluster2 fails: "error": "getManagedClusterResource results: missing ManagedClusterView conditions"

### ISSUE THREAD

**Issue by hatfieldbrian** [RamenDR/ramen#254]
configuration:
- 1 hub: hub
- 2 non-hub managed clusters: cluster1, cluster2

activity:

1. busybox-sample application initially deployed to cluster1 successfully
1. busybox-sample namespace created on cluster2
1. busybox-sample application failover to cluster2 fails - why?

DRPlacementControl reconcile loop log:
```
2021-09-01T03:26:28.496Z        INFO    controllers.DRPlacementControl  controller/controller.go:298    Entering reconcile loop {"DRPC": "busybox-sample/busybox-drpc"}
2021-09-01T03:26:28.496Z        INFO    controllers.DRPlacementControl  controllers/drplacementcontrol_controller.go:355        Validating schedule from DRPolicy
2021-09-01T03:26:28.496Z        INFO    controllers.DRPlacementControl  controllers/drplacementcontrol_controller.go:556        Getting User PlacementRule      {"placement": {"kind":"PlacementRule","name":"busybox-placement"}}
2021-09-01T03:26:28.496Z        INFO    controllers.DRPlacementControl  controllers/drplacementcontrol_controller.go:402        Starting to process placement
2021-09-01T03:26:28.496Z        INFO    controllers.DRPlacementControl  controllers/drplacementcontrol_controller.go:920        Process DRPC Placement
2021-09-01T03:26:28.496Z        INFO    controllers.DRPlacementControl  controllers/drplacementcontrol_controller.go:946        Entering runFailover    {"state": "Deployed"}
2021-09-01T03:26:28.496Z        INFO    controllers.DRPlacementControl  controllers/drplacementcontrol_controller.go:1116       Phase: Current 'Deployed'. Next 'FailingOver'
2021-09-01T03:26:28.496Z        INFO    controllers.DRPlacementControl  controllers/drplacementcontrol_controller.go:1959       Current Conditions '[{Available True 1 2021-08-31 22:23:15 +0000 UTC Deployed Condition type Available} {Reconciling True 1 2021-08-31 22:23:15 +0000 UTC Success Condition type Reconciling}]'
2021-09-01T03:26:28.496Z        INFO    controllers.DRPlacementControl  controllers/drplacementcontrol_controller.go:1959       Updated Conditions '[{Available False 3 2021-09-01 03:26:28.496941067 +0000 UTC m=+455465.827058694 FailingOver Condition type Available} {Reconciling False 3 2021-09-01 03:26:28.496942066 +0000 UTC m=+455465.827059681 Progressing Condition type Reconciling}]'
2021-09-01T03:26:28.496Z        INFO    controllers.DRPlacementControl  controllers/drplacementcontrol_controller.go:1132       Updating VRG ownedby MW busybox-drpc-busybox-sample-vrg-mw to secondary for cluster cluster1
2021-09-01T03:26:28.497Z        INFO    controllers.DRPlacementControl  controllers/drplacementcontrol_controller.go:1132       VRG busybox-drpc already secondary on this cluster cluster1
2021-09-01T03:26:28.497Z        INFO    controllers.DRPlacementControl  controllers/drplacementcontrol_controller.go:1145       runPlacementTask        {"cluster": "cluster2", "restorePVs": true}
2021-09-01T03:26:28.497Z        INFO    controllers.DRPlacementControl  controllers/drplacementcontrol_controller.go:1307       Checking whether PVs have been restored {"cluster": "cluster2"}
2021-09-01T03:26:28.497Z        INFO    controllers.DRPlacementControl  controllers/drplacementcontrol_controller.go:897        MCV condtions: []
2021-09-01T03:26:28.497Z        INFO    controllers.DRPlacementControl  controllers/drplacementcontrol_controller.go:1307       Failed to get VRG through MCV   {"error": "getManagedClusterResource results: missing ManagedClusterView conditions", "errorVerbose": "missing ManagedClusterView conditions\ngetManagedClusterResource results\ngithub.com/ramendr/ramen/controllers.(*DRPlacementControlReconciler).getManagedClusterResource\n\t/workspace/controllers/drplacementcontrol_controller.go:829\ngithub.com/ramendr/ramen/controllers.(*DRPlacementControlReconciler).getVRGFromManagedCluster\n\t/workspace/controllers/drplacementcontrol_controller.go:897\ngithub.com/ramendr/ramen/controllers.(*DRPCInstance).checkPVsHaveBeenRestored\n\t/workspace/controllers/drplacementcontrol_controller.go:1577\ngithub.com/ramendr/ramen/controllers.(*DRPCInstance).runPlacementTask\n\t/workspace/controllers/drplacementcontrol_controller.go:1307\ngithub.com/ramendr/ramen/controllers.(*DRPCInstance).switchToFailoverCluster\n\t/workspace/controllers/drplacementcontrol_controller.go:1145\ngithub.com/ramendr/ramen/controllers.(*DRPCInstance).runFailover\n\t/workspace/controllers/drplacementcontrol_controller.go:1110\ngithub.com/ramendr/ramen/controllers.(*DRPCInstance).processPlacement\n\t/workspace/controllers/drplacementcontrol_controller.go:946\ngithub.com/ramendr/ramen/controllers.(*DRPCInstance).startProcessing\n\t/workspace/controllers/drplacementcontrol_controller.go:920\ngithub.com/ramendr/ramen/controllers.(*DRPlacementControlReconciler).processAndHandleResponse\n\t/workspace/controllers/drplacementcontrol_controller.go:402\ngithub.com/ramendr/ramen/controllers.(*DRPlacementControlReconciler).reconcileDRPCInstance\n\t/workspace/controllers/drplacementcontrol_controller.go:395\ngithub.com/ramendr/ramen/controllers.(*DRPlacementControlReconciler).Reconcile\n\t/workspace/controllers/drplacementcontrol_controller.go:342\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.0-beta.4/pkg/internal/controller/controller.go:298\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.0-beta.4/pkg/internal/controller/controller.go:253\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.0-beta.4/pkg/internal/controller/controller.go:214\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1374"}
2021-09-01T03:26:28.497Z        INFO    controllers.DRPlacementControl  controllers/drplacementcontrol_controller.go:402        Process placement       {"error": "getManagedClusterResource results: missing ManagedClusterView conditions"}
2021-09-01T03:26:28.497Z        INFO    controllers.DRPlacementControl  controllers/drplacementcontrol_controller.go:395        Finished processing     {"Requeue?": true}
2021-09-01T03:26:28.497Z        INFO    controllers.DRPlacementControl  controllers/drplacementcontrol_controller.go:395        Requeing...
2021-09-01T03:26:28.497Z        INFO    controllers.DRPlacementControl  controller/controller.go:298    Exiting reconcile loop  {"DRPC": "busybox-sample/busybox-drpc"}
```

**Comment by hatfieldbrian** [comment]
Configuration was deleted and recreated today.  After hearing of issues Veera was experiencing with hub's `ocm-controller` deployment, I noticed it is unstable:
```
$ kubectl --context hub -n open-cluster-management get pods/ocm-controller-7c7fd888c8-27khn
NAME                              READY   STATUS             RESTARTS   AGE
ocm-controller-7c7fd888c8-27khn   0/1     CrashLoopBackOff   5          5m43s
bhatfiel@eyewall2:~$ kubectl --context hub -n open-cluster-management logs pods/ocm-controller-7c7fd888c8-27khn
W0915 03:16:00.009610       1 client_config.go:615] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
I0915 03:16:00.076826       1 leaderelection.go:243] attempting to acquire leader lease open-cluster-management/foundation-controller...
I0915 03:16:17.945008       1 leaderelection.go:253] successfully acquired lease open-cluster-management/foundation-controller
E0915 03:16:17.983322       1 reflector.go:138] k8s.io/client-go@v12.0.0+incompatible/tools/cache/reflector.go:167: Failed to watch *v1alpha1.PlacementDecision: failed to list *v1alpha1.PlacementDecision: placementdecisions.cluster.open-cluster-management.io is forbidden: User "system:serviceaccount:open-cluster-management:foundation-hub-sa" cannot list resource "placementdecisions" in API group "cluster.open-cluster-management.io" at the cluster scope
E0915 03:16:18.009495       1 server.go:211] Controller-runtime manager exited non-zero, failed to wait for imageregistry-controller caches to sync: no matches for kind "ManagedClusterImageRegistry" in version "imageregistry.open-cluster-management.io/v1alpha1"
failed to wait for imageregistry-controller caches to sync: no matches for kind "ManagedClusterImageRegistry" in version "imageregistry.open-cluster-management.io/v1alpha1"
```
This is part of the foundation operator added to support managed cluster view.  It's instability may be the reason for the "missing ManagedClusterView conditions" error.

**Comment by hatfieldbrian** [comment]
Re-ran failover test with recent ramen code, commit 319f678 plus pr #276, and same issue encountered:

```
2021-09-15T23:44:11.517Z        INFO    controllers.DRPlacementControl  controller/controller.go:298    Exiting reconcile loop  {"DRPC": "busybox-sample/busybox-drpc"}
2021-09-15T23:44:11.518Z        DEBUG   controller-runtime.manager.events       record/event.go:308     Warning {"object": {"kind":"DRPlacementControl","namespace":"busybox-sample","name":"busybox-drpc","uid":"8fed5fbb-d208-4eaf-ac42-d7e0380b42e0","apiVersion":"ramendr.openshift.io/v1alpha1","resourceVersion":"675211"}, "reason": "DRPCClusterSwitchFailed", "message": "getManagedClusterResource results: missing ManagedClusterView conditions"}
2021-09-15T23:44:12.158Z        INFO    controllers.DRPlacementControl  controller/controller.go:298    Entering reconcile loop {"DRPC": "busybox-sample/busybox-drpc"}
2021-09-15T23:44:12.165Z        INFO    controllers.DRPlacementControl  controllers/drplacementcontrol_controller.go:466        Validating schedule from DRPolicy
2021-09-15T23:44:12.165Z        INFO    controllers.DRPlacementControl  controllers/drplacementcontrol_controller.go:667        Getting User PlacementRule      {"placement": {"kind":"PlacementRule","name":"busybox-placement"}}
2021-09-15T23:44:12.165Z        INFO    controllers.DRPlacementControl  controllers/drplacementcontrol_controller.go:513        Starting to process placement
2021-09-15T23:44:12.165Z        INFO    controllers.DRPlacementControl  controllers/drplacementcontrol_controller.go:927        Process DRPC Placement
2021-09-15T23:44:12.165Z        INFO    controllers.DRPlacementControl  controllers/drplacementcontrol_controller.go:953        Entering runFailover    {"state": "Deployed"}
2021-09-15T23:44:12.165Z        INFO    controllers.DRPlacementControl  controllers/drplacementcontrol_controller.go:1117       Phase: Current 'Deployed'. Next 'FailingOver'
2021-09-15T23:44:12.165Z        INFO    controllers.DRPlacementControl  controllers/drplacementcontrol_controller.go:1969       Current Conditions '[{Available True 1 2021-09-15 23:41:27 +0000 UTC Deployed Condition type Available} {Reconciling True 1 2021-09-15 23:41:27 +0000 UTC Success Condition type Reconciling}]'
2021-09-15T23:44:12.165Z        INFO    controllers.DRPlacementControl  controllers/drplacementcontrol_controller.go:1969       Updated Conditions '[{Available False 2 2021-09-15 23:44:12.165593429 +0000 UTC m=+74214.672381297 FailingOver Condition type Available} {Reconciling False 2 2021-09-15 23:44:12.165594991 +0000 UTC m=+74214.672382847 Progressing Condition type Reconciling}]'
2021-09-15T23:44:12.165Z        INFO    controllers.DRPlacementControl  controllers/drplacementcontrol_controller.go:1133       Updating VRG ownedby MW busybox-drpc-busybox-sample-vrg-mw to secondary for cluster cluster1
2021-09-15T23:44:12.165Z        DEBUG   controller-runtime.manager.events       record/event.go:308     Warning {"object": {"kind":"DRPlacementControl","namespace":"busybox-sample","name":"busybox-drpc","uid":"8fed5fbb-d208-4eaf-ac42-d7e0380b42e0","apiVersion":"ramendr.openshift.io/v1alpha1","resourceVersion":"675211"}, "reason": "DRPCFailingOver", "message": "Failing over the application and VRG"}
2021-09-15T23:44:12.165Z        INFO    controllers.DRPlacementControl  controllers/drplacementcontrol_controller.go:1133       VRG busybox-drpc already secondary on this cluster cluster1
2021-09-15T23:44:12.166Z        INFO    controllers.DRPlacementControl  controllers/drplacementcontrol_controller.go:1146       runPlacementTask        {"cluster": "cluster2", "restorePVs": true}
2021-09-15T23:44:12.166Z        INFO    controllers.DRPlacementControl  controllers/drplacementcontrol_controller.go:1298       Checking whether PVs have been restored {"cluster": "cluster2"}
2021-09-15T23:44:12.167Z        INFO    MCV     controllers/drplacementcontrol_controller.go:177        MCV condtions: []       {"resouceName": "busybox-drpc"}
2021-09-15T23:44:12.167Z        INFO    controllers.DRPlacementControl  controllers/drplacementcontrol_controller.go:1298       Failed to get VRG through MCV   {"error": "getManagedClusterResource results: missing ManagedClusterView conditions", "errorVerbose": "missing ManagedClusterView conditions\ngetManagedClusterResource results\ngithub.com/ramendr/ramen/controllers.ManagedClusterViewGetterImpl.getManagedClusterResource\n\t/workspace/controllers/drplacementcontrol_controller.go:218\ngithub.com/ramendr/ramen/controllers.ManagedClusterViewGetterImpl.GetVRGFromManagedCluster\n\t/workspace/controllers/drplacementcontrol_controller.go:177\ngithub.com/ramendr/ramen/controllers.(*DRPCInstance).checkPVsHaveBeenRestored\n\t/workspace/controllers/drplacementcontrol_controller.go:1600\ngithub.com/ramendr/ramen/controllers.(*DRPCInstance).runPlacementTask\n\t/workspace/controllers/drplacementcontrol_controller.go:1298\ngithub.com/ramendr/ramen/controllers.(*DRPCInstance).switchToFailoverCluster\n\t/workspace/controllers/drplacementcontrol_controller.go:1146\ngithub.com/ramendr/ramen/controllers.(*DRPCInstance).runFailover\n\t/workspace/controllers/drplacementcontrol_controller.go:1111\ngithub.com/ramendr/ramen/controllers.(*DRPCInstance).processPlacement\n\t/workspace/controllers/drplacementcontrol_controller.go:953\ngithub.com/ramendr/ramen/controllers.(*DRPCInstance).startProcessing\n\t/workspace/controllers/drplacementcontrol_controller.go:927\ngithub.com/ramendr/ramen/controllers.(*DRPlacementControlReconciler).processAndHandleResponse\n\t/workspace/controllers/drplacementcontrol_controller.go:513\ngithub.com/ramendr/ramen/controllers.(*DRPlacementControlReconciler).reconcileDRPCInstance\n\t/workspace/controllers/drplacementcontrol_controller.go:506\ngithub.com/ramendr/ramen/controllers.(*DRPlacementControlReconciler).Reconcile\n\t/workspace/controllers/drplacementcontrol_controller.go:453\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.0-beta.4/pkg/internal/controller/controller.go:298\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.0-beta.4/pkg/internal/controller/controller.go:253\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.0-beta.4/pkg/internal/controller/controller.go:214\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1374"}
2021-09-15T23:44:12.167Z        INFO    controllers.DRPlacementControl  controllers/drplacementcontrol_controller.go:513        Process placement       {"error": "getManagedClusterResource results: missing ManagedClusterView conditions"}
2021-09-15T23:44:12.168Z        INFO    controllers.DRPlacementControl  controllers/drplacementcontrol_controller.go:506        Finished processing     {"Requeue?": true}
2021-09-15T23:44:12.168Z        INFO    controllers.DRPlacementControl  controllers/drplacementcontrol_controller.go:506        Requeing...
2021-09-15T23:44:12.167Z        DEBUG   controller-runtime.manager.events       record/event.go:308     Warning {"object": {"kind":"DRPlacementControl","namespace":"busybox-sample","name":"busybox-drpc","uid":"8fed5fbb-d208-4eaf-ac42-d7e0380b42e0","apiVersion":"ramendr.openshift.io/v1alpha1","resourceVersion":"675211"}, "reason": "DRPCClusterSwitchFailed", "message": "getManagedClusterResource results: missing ManagedClusterView conditions"}
2021-09-15T23:44:12.168Z        INFO    controllers.DRPlacementControl  controller/controller.go:298    Exiting reconcile loop  {"DRPC": "busybox-sample/busybox-drpc"}
```

then the next reconcile loop it proceeds further:

```
2021-09-15T23:44:13.096Z        INFO    controllers/drplacementcontrol_controller.go:398        Filtering MCV (busybox-drpc-busybox-sample-vrg-mcv/cluster2)
2021-09-15T23:44:13.096Z        INFO    controllers/drplacementcontrol_controller.go:398        Filtering MCV (busybox-drpc-busybox-sample-vrg-mcv/cluster2)
2021-09-15T23:44:13.097Z        INFO    controllers.DRPlacementControl  controller/controller.go:298    Entering reconcile loop {"DRPC": "busybox-sample/busybox-drpc"}
2021-09-15T23:44:13.105Z        INFO    controllers.DRPlacementControl  controllers/drplacementcontrol_controller.go:466        Validating schedule from DRPolicy
2021-09-15T23:44:13.105Z        INFO    controllers.DRPlacementControl  controllers/drplacementcontrol_controller.go:667        Getting User PlacementRule      {"placement": {"kind":"PlacementRule","name":"busybox-placement"}}
2021-09-15T23:44:13.106Z        INFO    controllers.DRPlacementControl  controllers/drplacementcontrol_controller.go:513        Starting to process placement
2021-09-15T23:44:13.106Z        INFO    controllers.DRPlacementControl  controllers/drplacementcontrol_controller.go:927        Process DRPC Placement
2021-09-15T23:44:13.106Z        INFO    controllers.DRPlacementControl  controllers/drplacementcontrol_controller.go:953        Entering runFailover    {"state": "Deployed"}
2021-09-15T23:44:13.106Z        INFO    controllers.DRPlacementControl  controllers/drplacementcontrol_controller.go:1117       Phase: Current 'Deployed'. Next 'FailingOver'
2021-09-15T23:44:13.106Z        INFO    controllers.DRPlacementControl  controllers/drplacementcontrol_controller.go:1969       Current Conditions '[{Available True 1 2021-09-15 23:41:27 +0000 UTC Deployed Condition type Available} {Reconciling True 1 2021-09-15 23:41:27 +0000 UTC Success Condition type Reconciling}]'
2021-09-15T23:44:13.106Z        INFO    controllers.DRPlacementControl  controllers/drplacementcontrol_controller.go:1969       Updated Conditions '[{Available False 2 2021-09-15 23:44:13.106163421 +0000 UTC m=+74215.612951380 FailingOver Condition type Available} {Reconciling False 2 2021-09-15 23:44:13.106166474 +0000 UTC m=+74215.612954365 Progressing Condition type Reconciling}]'
2021-09-15T23:44:13.106Z        INFO    controllers.DRPlacementControl  controllers/drplacementcontrol_controller.go:1133       Updating VRG ownedby MW busybox-drpc-busybox-sample-vrg-mw to secondary for cluster cluster1
2021-09-15T23:44:13.106Z        INFO    controllers.DRPlacementControl  controllers/drplacementcontrol_controller.go:1133       VRG busybox-drpc already secondary on this cluster cluster1
2021-09-15T23:44:13.106Z        INFO    controllers.DRPlacementControl  controllers/drplacementcontrol_controller.go:1146       runPlacementTask        {"cluster": "cluster2", "restorePVs": true}
2021-09-15T23:44:13.106Z        INFO    controllers.DRPlacementControl  controllers/drplacementcontrol_controller.go:1298       Checking whether PVs have been restored {"cluster": "cluster2"}
2021-09-15T23:44:13.106Z        INFO    MCV     controllers/drplacementcontrol_controller.go:177        MCV condtions: [{Processing True 0 2021-09-15 23:44:13 +0000 UTC GetResourceProcessing Watching resources successfully}]        {"resouceName": "busybox-drpc"}
2021-09-15T23:44:13.107Z        INFO    controllers.DRPlacementControl  controllers/drplacementcontrol_controller.go:1298       Waiting for PVs to be restored  {"cluster": "cluster2"}
2021-09-15T23:44:13.107Z        INFO    controllers.DRPlacementControl  controllers/drplacementcontrol_controller.go:1146       Checked whether PVs have been restored  {"Yes?": false}
2021-09-15T23:44:13.107Z        INFO    controllers.DRPlacementControl  controllers/drplacementcontrol_controller.go:513        Process placement       {"error": "Waiting for PV restore to complete)"}
2021-09-15T23:44:13.107Z        INFO    controllers.DRPlacementControl  controllers/drplacementcontrol_controller.go:506        Finished processing     {"Requeue?": true}
2021-09-15T23:44:13.107Z        INFO    controllers.DRPlacementControl  controllers/drplacementcontrol_controller.go:506        Requeing...
2021-09-15T23:44:13.107Z        INFO    controllers.DRPlacementControl  controller/controller.go:298    Exiting reconcile loop  {"DRPC": "busybox-sample/busybox-drpc"}
2021-09-15T23:44:13.107Z        DEBUG   controller-runtime.manager.events       record/event.go:308     Warning {"object": {"kind":"DRPlacementControl","namespace":"busybox-sample","name":"busybox-drpc","uid":"8fed5fbb-d208-4eaf-ac42-d7e0380b42e0","apiVersion":"ramendr.openshift.io/v1alpha1","resourceVersion":"675211"}, "reason": "DRPCFailingOver", "message": "Failing over the application and VRG"}
2021-09-15T23:44:13.107Z        DEBUG   controller-runtime.manager.events       record/event.go:308     Warning {"object": {"kind":"DRPlacementControl","namespace":"busybox-sample","name":"busybox-drpc","uid":"8fed5fbb-d208-4eaf-ac42-d7e0380b42e0","apiVersion":"ramendr.openshift.io/v1alpha1","resourceVersion":"675211"}, "reason": "DRPCClusterSwitchFailed", "message": "Waiting for PV restore to complete)"}
```

Since latest run got past "getManagedClusterResource results: missing ManagedClusterView conditions", this issue can be closed.

**Comment by hatfieldbrian** [comment]
Determined reason for PV download failure was minio-on-hub failed to satisfy persistent volume claim with a rook-ceph-block storage class persistent volume.  Pull request coming that deploys an s3store on each managed cluster instead of one on the hub.
